packages/core/diagnostic/src/diagnostic.js

```diff
@@ -213,7 +212,7 @@ export function generateJSONCodeHighlights(
   // json-source-map doesn't support a tabWidth option (yet)
   let map = jsonMap.parse(code.replace(/\t/g, ' '));
   return ids.map(({key, type, message}) => {
-    let pos = nullthrows(map.pointers[key]);
+    let pos = map.pointers[key];
```

Comment:
I'm afraid that Flow cannot always be trusted 😬 

`item` isn't a number in this case, but Flow doesn't complain.
```
type Pointers =  {|[key: string]: number|};

let x: Pointers = {b: 1};

let item = x['a'];
(item: number);
```

https://flow.org/try/#0C4TwDgpgBACg9gSwHbAgJwM5QLxSgbwB8BtAawhAC4oNg1kBzAXWqQFcBbAI3UIF8A3ACghAGwjAoAD2rxkqTDgKCR4yQlQclU4gHIAhrqbCAFBogdWnHmgCUAoA

In our situation here, the `pos` parameter of `getJSONSourceLocation` could suddenly be `undefined`, although Flow says it's an object.

Reply:
@mischnic Good catch! I've added `nullthrows` back.